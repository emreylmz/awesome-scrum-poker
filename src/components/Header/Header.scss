@import "../../styles/colors";
@import "../../styles/metrics";

.header {
  flex: 1;
  align-items: flex-start;
  height: 100px;
  margin: $default-margin;
  &__title {
    color: $primary-color
  }
}
