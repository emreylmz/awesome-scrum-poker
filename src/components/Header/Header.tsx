import React from 'react'

interface Props {

}

interface State {

}

class Header extends React.PureComponent<Props, State> {
  render() {
    return (
      <h1></h1>
    );
  }
}

export type HeaderProps = Props;
export type HeaderState = State;
export default Header;
