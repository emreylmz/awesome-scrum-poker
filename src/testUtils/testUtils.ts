export const emptyTestFunc = (): void => void 0
