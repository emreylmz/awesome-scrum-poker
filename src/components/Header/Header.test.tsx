import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import '../../testUtils/setupTests'
import Header, { HeaderProps } from './Header'

type DoneCallback = jest.DoneCallback

const defaultTitle = 'My Header'

const getWrapper = (props?: HeaderProps): ShallowWrapper => {
  return shallow(<Header title={defaultTitle} {...props} />)
}

describe('Header UI', () => {
  it('should render correctly', (done: DoneCallback) => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it('should have title', (done: DoneCallback) => {
    const wrapper = getWrapper()
    const titleWrapper = wrapper.find('.header__title')
    expect(titleWrapper.text()).toEqual(defaultTitle)
    done()
  })
})
