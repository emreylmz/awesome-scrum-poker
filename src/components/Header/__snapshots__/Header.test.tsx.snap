// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI should render correctly 1`] = `ShallowWrapper {}`;
